pkg/cmd/release/create/create.go

```diff
@@ -268,12 +279,20 @@ func createRun(opts *CreateOptions) error {
 		}
 	}
 
+	if opts.Draft && len(opts.DiscussionCategoryName) > 0 {
+		return fmt.Errorf(
+			"%s Discussions not supported with draft releases",
+			opts.IO.ColorScheme().FailureIcon(),
+		)
+	}
+
 	params := map[string]interface{}{
-		"tag_name":   opts.TagName,
-		"draft":      opts.Draft,
-		"prerelease": opts.Prerelease,
-		"name":       opts.Name,
-		"body":       opts.Body,
+		"tag_name":                 opts.TagName,
+		"draft":                    opts.Draft,
+		"prerelease":               opts.Prerelease,
+		"name":                     opts.Name,
+		"body":                     opts.Body,
+		"discussion_category_name": opts.DiscussionCategoryName,
```

Comment:
Should we only add this param if the flag was set? That way we avoid having to change a bunch of old tests